lib/runner/extensions/request.command.js

```diff
@@ -14,43 +14,76 @@ var _ = require('lodash'),
      */
     resolveFiles = function (request, resolver, triggers, callback) {
         var mode = _.get(request, 'body.mode'),
-            formParams = _.get(request, 'body.formdata'),
-            fileData = _.get(request, 'body.file');
+            sourceReadable = _.isFunction(resolver && resolver.createReadStream),
+            data = mode && _.get(request, 'body.' + mode);
 
-        if (!mode) { return callback(null, request); }
+        // if there is no mode specified, or no data for the specified mode we cannot resolve anything!
+        // @note that if source is not readable, there is no point reading anything, yet we need to warn that file
+        // upload was not done. hence we will have to proceed even without an unreadable source
+        if (!data) { // we do not need to check `mode` here since false mode returns no `data`
+            return callback(null, request);
+        }
 
-        if (mode === 'formdata') {
-            return async.reduce(formParams.all(), [], function (list, param, cb) {
-                var isFile = (param.type === 'file');
+        // in this block, we simply use async.waterfall to ensure that all form of file reading is async. essentially,
+        // we first determine the data mode and based on it pass the waterfall functions.
+        async.waterfall([async.constant(data), {
+            // form data parsing simply "enriches" all form parameters having file data type by replacing / setting the
+            // value as a read stream
+            formdata: function (formdata, next) {
+                // ensure that we only process the file type
+                async.eachSeries(_.filter(formdata.all(), {type: 'file'}), function (formparam, next) {
+                    var stream;
+
+                    try {
+                        sourceReadable && formparam.src && (stream = resolver.createReadStream(formparam.src));
+                    }
+                    // eslint-disable-next-line no-empty
+                    catch (e) { } // we do not need to forward error here since next line we throw warning
+
+                    if (!stream) { // lack of stream also determines lack of `formparam.src` and source being readable
+                        // @todo - get cursor here.
+                        triggers.console({}, 'warn', 'unable to load form file for upload: "' + formparam.src + '"');
```

Comment:
Will do it as part of the task on later revision. I don't what to increase the scope of this PR further.